### PR TITLE
ENG-1634: Add --build to state publish: build, encrypt, wrap, upload

### PR DESCRIPTION
**Pull request by mitchell-as**
[ENG-1634: Add --build to state publish: build, encrypt, wrap, upload](https://activestatef.atlassian.net/browse/ENG-1634)

The orchestration that ties the private ingredient publish side together ([ENG-1563](https://activestatef.atlassian.net/browse/ENG-1563)). `state publish --build <srcdir>` packages a customer's source on their own machine, locks it with the organization's key, and uploads only the locked result — the hosting service never sees the unlocked content. If the key is unavailable it fails before uploading anything, and the plaintext only ever lives in a temporary directory. Publishing without `--build` behaves exactly as before.

**Notes:** A private publish isn't genesis-stamped yet — that depends on a platform-side flag (ENG-1641), so it's left as a TODO. And the encrypted payload carries the wheel file as-is; turning it into an installed package is the consume side's job.

**Base branch:** targets `mitchell/eng-1633` so the diff is only this change; GitHub will retarget it to `version/0-48-1-RC2` once the upstream PRs land.

Tested for the wrapped-artifact contents, round-trip decryption, plaintext cleanup, and metadata resolution.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[ENG-1563]: https://activestatef.atlassian.net/browse/ENG-1563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/runners/publish/build.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+func (r *Runner) generateEncryptedArtifact(params *Params) (cleanup func(), rerr error) {
+	if params.Filepath != "" {
+		return nil, locale.NewInputError("err_publish_build_and_file", "The '[ACTIONABLE]--build[/RESET]' flag cannot be combined with a source archive filepath.")
+	}
+	if r.project == nil {
+		return nil, locale.NewInputError("err_publish_build_no_project", "The '[ACTIONABLE]--build[/RESET]' flag requires a project so the organization can be determined.")
+	}
+	if params.Namespace != "" && !strings.HasPrefix(params.Namespace, "private/") {
+		return nil, locale.NewInputError("err_publish_build_namespace", "The '[ACTIONABLE]--build[/RESET]' flag requires a '[ACTIONABLE]private/[/RESET]' namespace.")
+	}
```
[the remark this answers is not in the document]
Comment (mitchell-as) [reply internal/runners/publish/build.go]: Fixed in 4090ca1d0. The namespace is now validated against the project's own org — private/<owner> or a sub-namespace of it — on the resolved value, so it covers the --namespace flag, a --meta file's namespace, and the default, not just the flag's private/ prefix. This closes the wrong-org case where the artifact is encrypted under this org's key but published under another. Added a unit test for the boundary cases (including private/myorg2 not matching private/myorg).

File: internal/runners/publish/publish.go [hunk 2]
```diff
@@ -87,14 +92,22 @@ func (r *Runner) Run(params *Params) error {
 		return locale.NewInputError("err_auth_required")
 	}
 
+	if params.Build != "" {
+		cleanup, err := r.generateEncryptedArtifact(params) // note: this function also mutates params
+		if err != nil {
+			return errs.Wrap(err, "Could not build private ingredient")
+		}
+		defer cleanup() // remove artifact.tar.gz and temporary build dir
```
[the remark this answers is not in the document]
Comment (mitchell-as) [reply internal/runners/publish/publish.go]: Fixed in 4090ca1d0 — the comment now reads "remove the temporary build directory", since cleanup removes the whole temp dir rather than a single archive.

File: internal/runners/publish/build_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	f, err := os.Open(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	gz, err := gzip.NewReader(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tr := tar.NewReader(gz)
```
[the remark this answers is not in the document]
Comment (mitchell-as) [reply internal/runners/publish/build_test.go:46]: Fixed in 4090ca1d0 — added defer gz.Close() in readTarGz.

**Comment by mitchell-as** [comment]
macOS and Windows test failures are sporadic/known and are unrelated to this PR.
